### PR TITLE
Fixed the reporting of the discriminator loss

### PR DESCRIPTION
**Pull request by vincentpierre**
### Proposed change(s)

Modify what is reported under GAIL Discriminator Loss.

The inconsistencies between TF and torch are due to our reporting
The loss will still be different between torch and tf since tf reports the discriminator loss + gradient magnitude loss.
The gradient magnitude loss is not identical between torch and tf but there seems to be little difference in training

### Types of change(s)

- [ ] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Breaking change
- [ ] Documentation update
- [ ] Other (please describe)

### Checklist
- [ ] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/master/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [ ] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/master/docs) (if applicable)
- [ ] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Migrating.md) (if applicable)

### CODE REVIEW

File: ml-agents/mlagents/trainers/torch/components/reward_providers/gail_reward_provider.py [hunk 1]
```diff
@@ -177,41 +167,55 @@ def compute_loss(
...
+        )
+        stats_dict["Policy/GAIL Expert Estimate"] = (
+            expert_estimate.mean().detach().cpu().numpy()
+        )
+        discriminator_loss = -(
+            torch.log(expert_estimate + self.EPSILON)
+            + torch.log(1.0 - policy_estimate + self.EPSILON)
         ).mean()
-        kl_loss: Optional[torch.Tensor] = None
+        stats_dict["Losses/GAIL Discriminator Loss"] = (
```
Comment (andrewcoh) [inline ml-agents/mlagents/trainers/torch/components/reward_providers/gail_reward_provider.py:188]: Should we name the reported things the same as they are in TF even though this is more descriptive? Maybe change the TF reporting from "Losses/GAIL Loss"?

**Comment by andrewcoh** [comment]
> LGTM, though I'll wait for Andrew to say he sees the right plots

but how will he know if he sees the right plots

File: ml-agents/mlagents/trainers/torch/components/reward_providers/gail_reward_provider.py (hunk 1 above)
Comment (vincentpierre) [reply ml-agents/mlagents/trainers/torch/components/reward_providers/gail_reward_provider.py:188]:
```suggestion
        stats_dict["Losses/GAIL Loss"] = (
```

**Comment by vincentpierre** [comment]
The plots will not be identical because TF adds the gradient magnitude loss to the GAIL loss (and I do not)
It also seems that my computation of the gradient magnitude loss is different in torch than in TensorFlow and I can't figure out why. (I suspect it is du to my call to `torch.autograd.grad`
